### PR TITLE
[4078] Loader missing on checkout page fix

### PR DESCRIPTION
**Pull request by Mohamed-halim-sw**
**Related issue(s):**
* Fixes scandipwa/scandipwa/issues/4078

**Problem:**
* summary block loader is missing while logging in and displays "0 items" instead.

**In this PR:**
* Cart.action
   	* line 18 added action type,
	* line 85-89 added updateIsLoading function,
	
* Cart.reducer
	* line 15 import UPDATE_TOTALS_IS_LOADING TYPE,
	* line 81-86 added UPDATE_TOTAL_IS_LOADING case and it's return,
	* line 44 added isloading:false in updatecarttotals method return
	* line 66 added isloading:false in initial state
	
* Cart.dispatcher 
	* line 13 import updateIsLoading
	* line 256 to 258 added _updateCartIsLoading function
	* line 54 call dispatch on updatecartisloading set to true
	* line 41 call dispatch on updatecartisloading set to false
	
* CheckoutOrderSummary.container
	* line 33 added isloading prop
	
* CheckoutOrderSummary.component
	* line 18 import loader component
	* line 47 isloading prop
	* line 52 isloading defaultProp
	* line 328 added renderloader method
	* line 355 added a condition to render loader if isloading is true

### CODE REVIEW

**Comment by Elena-Evsa** [comment]
Issue 1: Make the frame sized like on screenshot. Move loader to the center of frame
image

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 1]
```diff
@@ -322,12 +325,19 @@ export class CheckoutOrderSummary extends PureComponent {
...
+        return <Loader isLoading={ isLoading } />;
+    }
+
     renderExpandableContent() {
         return (
             <ExpandableContent
               heading={ __('Summary') }
               mix={ { block: 'CheckoutOrderSummary', elem: 'ExpandableContent' } }
             >
+                { this.renderLoader() }
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:340]: In this case can just pass component loader without making a function.

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js [hunk 2]
```diff
@@ -352,8 +362,10 @@ export class CheckoutOrderSummary extends PureComponent {
     }
 
     render() {
+        const { isLoading } = this.props;
         return (
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.component.js:366]: Space before return

File: packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.container.js [hunk 3]
```diff
@@ -23,12 +23,15 @@ import CheckoutOrderSummary from './CheckoutOrderSummary.component';
...
+
     cartDisplayConfig: state.ConfigReducer.cartDisplayConfig,
     cartSubtotal: getCartSubtotal(state),
     cartSubtotalSubPrice: getCartSubtotalSubPrice(state),
     cartShippingPrice: getCartShippingPrice(state),
     cartShippingSubPrice: getCartShippingSubPrice(state),
-    cartTotalSubPrice: getCartTotalSubPrice(state)
+    cartTotalSubPrice: getCartTotalSubPrice(state),
+    isLoading: state.CartReducer.isLoading
+
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/component/CheckoutOrderSummary/CheckoutOrderSummary.container.js:34]: redundant space

File: packages/scandipwa/src/route/Checkout/Checkout.component.js [hunk 4]
```diff
@@ -367,6 +368,7 @@ export class Checkout extends PureComponent {
               renderCmsBlock={ () => this.renderPromo(true) }
               showItems
             />
+
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/Checkout/Checkout.component.js:371]: Same here, no need space

File: packages/scandipwa/src/route/Checkout/Checkout.component.js [hunk 5]
```diff
@@ -358,6 +358,7 @@ export class Checkout extends PureComponent {
         }
 
         return (
+
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/route/Checkout/Checkout.component.js:361]: also here

File: packages/scandipwa/src/store/Cart/Cart.dispatcher.js [hunk 6]
```diff
@@ -51,7 +52,7 @@ export class CartDispatcher {
             } = await fetchMutation(CartQuery.getCreateEmptyCartMutation());
 
             setGuestQuoteId(quoteId);
-            this._updateCartData({}, dispatch);
+            this._updateCartIsLoading(true, dispatch);
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/store/Cart/Cart.dispatcher.js:55]: No need for function, can directly call dispatch update is loading.

File: packages/scandipwa/src/store/Cart/Cart.dispatcher.js [hunk 7]
```diff
@@ -38,6 +38,7 @@ export class CartDispatcher {
                 )
             );
 
+            this._updateCartIsLoading(false, dispatch);
             return this._updateCartData(cartData, dispatch);
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/store/Cart/Cart.dispatcher.js:42]: Space before return

File: packages/scandipwa/src/store/Cart/Cart.reducer.js [hunk 8]
```diff
@@ -78,6 +79,12 @@ export const CartReducer = (
         return updateCartTotals(action, state);
     case UPDATE_SHIPPING_PRICE:
         return updateShippingPrice(action, state);
+    case UPDATE_TOTALS_IS_LOADING:
+        const { isLoading } = action;
+        return {
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/store/Cart/Cart.reducer.js:84]: space before return

File: packages/scandipwa/src/store/Cart/Cart.action.js [hunk 9]
```diff
@@ -80,3 +81,9 @@ export const applyCouponToCart = (couponCode) => ({
 export const removeCouponFromCart = () => ({
     type: REMOVE_COUPON_FROM_CART
 });
+
+/** @namespace Store/Cart/Action/updateIsLoading */
+export const updateIsLoading = (isLoading) => ({
+    type: UPDATE_TOTALS_IS_LOADING,
+    isLoading
+});
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/src/store/Cart/Cart.action.js:89]: make semantically correct name of function, like updateIsLoadingCart, same for type

**Review by AleksandrsKondratjevs — CHANGES_REQUESTED** [review]

File: packages/scandipwa/package.json [hunk 10]
```diff
@@ -473,5 +473,5 @@
     "publishConfig": {
         "access": "public"
     },
-    "proxy": "https://demo100-ors-1588667385-csa-hcx.scandipwa.cloud/"
+    "proxy": "https://scandipwapmrev.indvp.com/"
```
Comment (AleksandrsKondratjevs) [inline packages/scandipwa/package.json:476]: Get it back

File: packages/scandipwa/src/store/Cart/Cart.reducer.js (hunk 8 above)
Comment (Mohamed-halim-sw) [reply packages/scandipwa/src/store/Cart/Cart.reducer.js:84]: do i need to add space?

File: packages/scandipwa/package.json (hunk 10 above)
Comment (Mohamed-halim-sw) [reply packages/scandipwa/package.json:476]: get it to demo100?

Comment (AleksandrsKondratjevs) [reply packages/scandipwa/package.json:476]: yes

**Review by AleksandrsKondratjevs — APPROVED** [review]

**Pull request merged**